tests/webp-uploads-test.php

```diff
@@ -0,0 +1,39 @@
+<?php
+/**
+ * Tests for webp-uploads module.
+ *
+ * @package performance-lab
+ */
+
+class WebP_Uploads_Tests extends WP_UnitTestCase {
+
+	/**
+	 * Test if webp-uploads is loaded and filter applied when system supports WebP.
+	 */
+	function test_webp_uploads_is_loaded() {
+		// Test before enabling.
+		$output_format = apply_filters( 'image_editor_output_format', array(), '', 'image/jpeg' );
+		$this->assertEquals( array(), $output_format );
+
+		// Activate the module.
+		$new_value = array(
+			'webp-uploads' => array( 'enabled' => true ),
+		);
+		update_option( PERFLAB_MODULES_SETTING, $new_value );
+		perflab_load_active_modules();
+
+		if ( ! wp_image_editor_supports( array( 'mime_type' => 'image/webp' ) ) ) {
```

Comment:
would be ideal to run this test against several PHP versions, some that support WebP and some that don't. That way we can validate that the filter is only applied when the system supports WebP.

this check may be overly defensive as I think we do a similar check in core before actually switching, but I thik still worth having.

Reply:
Agreed. Right now the test suite only runs on latest WordPress and PHP 7.4, maybe let's open a separate issue to evaluate what versions we should run on? We also need to decide in general which WordPress and PHP versions the plugin should support overall, so that could be discussed in the same issue.